pilot/pkg/model/sidecar.go

```diff
@@ -556,3 +520,44 @@ func (ilw *IstioEgressListenerWrapper) selectServices(services []*Service, confi
 	}
 	return filteredServices
 }
+
+func matchingServices(importedHosts []host.Name, service *Service, ilw *IstioEgressListenerWrapper) (bool, []*Service) {
+	// If a listener is defined with port, we should match services with port.
+	needsPortMatch := ilw.IstioListener != nil && ilw.IstioListener.Port != nil
```

Comment:
@rshriram once this PR merges, we can add additional condition here to check for `Http_Proxy` protocol in Port and skip port matching.

Reply:
Still have to change routes - but just pushed get your initial thoughts